### PR TITLE
switched to google formatter

### PR DESCRIPTION
**Pull request by matthiasbalke**
This PR adds the google formatter using the spotless gradle plugin.
This way a centralized code style can be guaranteed.

* The codebase must be formattet once in a seperate commit.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## How can I contribute?
+
+Any kind of contribution is welcome as a pull request.
+If you are unsure on how to fork a repository and then create a pull 
+request from your fork, please read [this blog post](http://www.reflectoring.io/hacks/github-fork-and-pull/)
+for a quick guide.
+
+### Pick an Issue
+You can pick any issue from the [issue tracker](https://github.com/reflectoring/infiniboard/issues). 
+I try to document all feature ideas, tasks and bugs as issues in the tracker.
```
Comment (RudiKlassen) [inline CONTRIBUTING.md]: we´ll stand behind you

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Issues marked with the tag ["up-for-grabs"](https://github.com/reflectoring/infiniboard/issues?q=is%3Aissue+is%3Aopen+label%3Aup-for-grabs) are especially documented so that 
+new contributors should get an idea of what is to do. You might want to
+start with one of those tasks. You can just as well pick any other task, though.
+If you have any questions, get in touch.
+
+### Submit an Idea
+If you have an idea, submit it to the [issue tracker](https://github.com/reflectoring/infiniboard/issues)
+for discussion. I will let you know my thoughts. If you want to develop your idea
```
Comment (RudiKlassen) [inline CONTRIBUTING.md]: and i, .. or just we:)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+yourself, you can do so and submit a pull request.
+
```
Comment (RudiKlassen) [inline CONTRIBUTING.md]: We have a wiki-page for discussions and ideas. Maybe we should provide the link in this guide too.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+Below are some hints on conventions used in this project. If you are unsure about
+any, just get in touch. During a pull request review, we will also check these
+conventions. Fear not, the worst thing that may happen if you do not follow them
+is that we might propose some changes to a pull request you submitted.
+
+### Styleguide
+For java code style, the gradle plugin ["spotless"](https://github.com/diffplug/spotless) has been included in the build to enforce the
+[Google Java Code style](https://google.github.io/styleguide/javaguide.html). Builds will fail if 
+the code does not comply to the style guide. To apply the style guide simply call 
+`gradlew spotlessApply`. Only after applying the style guide should you push your changes.
```
Comment (RudiKlassen) [inline CONTRIBUTING.md]: Should we add this information to our readme?

File: CONTRIBUTING.md (hunk 2 above)
Comment (matthiasbalke) [inline CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (matthiasbalke) [inline CONTRIBUTING.md]: We have a link to the contribution guide in our readme, I think this is enough.

File: CONTRIBUTING.md (hunk 3 above)
Comment (matthiasbalke) [inline CONTRIBUTING.md]: I think we have to rework the wiki pages before linking them to public pages.

File: CONTRIBUTING.md (hunk 4 above)
Comment (RudiKlassen) [inline CONTRIBUTING.md]: ok

**Pull request merged**